src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts

```diff
@@ -247,7 +247,7 @@ export class HostsComponent extends ListWithDetails implements OnInit {
             error.error['detail'].includes('WARNING') &&
             !error.error['detail'].includes('It is NOT safe to stop') &&
             !error.error['detail'].includes('ALERT') &&
-            !error.error['detail'].includes('unable to stop')
+            !error.error['detail'].includes('unsafe to stop')
```

Comment:
I'm a bit concerned about the fact that we rely on parsing so many literals to distinguish different error conditions. The outcome of the back-end calls should be clear and unequivocal (either OK or ERROR; if ERROR, then use ERROR codes/context).  

Can't we make the orchestrator return different error codes on each condition? Besides this, how does a REST-only user know about this (if it's UI-only handling)?

Reply:
@epuertat These conditions are there for UI to decide when the force maintenance modal should show. Unfortunately I couldn't find an alternative for this when the time I did the work. @jmolmo Do you think the orchestrator can return different error codes for each condition?
Also, for REST-only api this is not a concern. There is a `force` flag for putting the host in maintenance. Even if the user tries to force a host which can't be put into maintenance mode, the error message will be displayed.